shesmu-pluginapi/src/main/java/ca/on/oicr/gsi/shesmu/plugin/cache/KeyValueCache.java

```diff
@@ -169,4 +169,30 @@ public final void ttl(int ttl) {
     this.ttl = ttl;
     ttlValue.labels(name).set(ttl);
   }
+
+  public String toString() {
+    StringBuilder writeOut = new StringBuilder();
+    writeOut
+        .append("KeyValueCache of types ")
+        .append(getKType())
+        .append(" and ")
+        .append(getVType())
+        .append(" with CACHES = ")
+        .append(CACHES.entrySet())
+        .append(", maxCount = ")
+        .append(maxCount)
+        .append(", name = ")
+        .append(name)
+        .append(", recordFactory = ")
+        .append(recordFactory)
+        .append(", records = ")
+        .append(records.entrySet())
+        .append(", ttl = ")
+        .append(ttl);
+    return writeOut.toString();
+  }
+
+  protected abstract String getKType();
+
+  protected abstract String getVType();
```

Comment:
`getKeyType` and `getValueType` would be more clear, but you're also basically defining these at the class level. Would it be more useful to just log the class name in `toString` rather than the type parameters? That would also be more concise and impossible to mess up when implementing a new subclass

Reply:
I designed it this way to minimize repeat code, if I moved toString to the class level then there would be ~20 lines of repeat code in each child class of KeyValueCache

Reply:
I meant that "KType" and "VType" are being defined at the class level right now - the return value is a static String for all implementations. This means the class name is just as useful, if not more so since 2 classes could have the same types. In `KeyValueCache#toString` you could just log `getClass()` (or one of its String properties) instead of needing `getKType` and `getVType` methods

Reply:
Didn't realize this would get the actual class name instead of the abstract class's name, thanks, fixed